.github/workflows/populate-preview-deployment-id.yaml

```diff
@@ -0,0 +1,29 @@
+name: Populate Preview Deployment ID
+
+on:
+  pull_request:
+    branches:
+      - master
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: tibdex/github-app-token@v1
+        id: generate-token
+        with:
+          app_id: ${{ secrets.HMCTS_GITHUB_EXUI_APP_ID }}
+          private_key: ${{ secrets.HMCTS_GITHUB_EXUI_PRIVATE_KEY }}
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ steps.generate-token.outputs.token }}
+      - name: Populate preview deployment ID
+        run: |
+          sed -i 's/PREVIEW_DEPLOYMENT_ID:.*/PREVIEW_DEPLOYMENT_ID: exui-preview-deployment-${{ github.event.number }}/' charts/xui-webapp/values.preview.template.yaml
```

Comment:
Maybe be slightly more specific on this pattern, so that it only matches at the the start of a line, something like 
s/^ *PREVIEW_DEPLOYMENT_ID:.*/

Reply:
We will want to remove any preview deployment id that has been left behind by an old PR - as not everyone will remember to clean up so I wanted to replace the whole line.